toolbox/api/datagalaxy_api_dictionary.py

```diff
@@ -0,0 +1,161 @@
+            headers = {'Authorization': f"Bearer {self.access_token}"}
+            response = requests.get(f"{self.url}/structures", params=params, headers=headers)
+            code = response.status_code
+            body_json = response.json()
+            if code == 200:
+                logging.info(
+                    f'list_structures - {len(body_json["results"])} structures found on '
+                    f'workspace: {workspace_name}')
+                result = []
+                result = result + body_json['results']
+                next_page = body_json["next_page"]
+                while next_page is not None:
+                    headers = {'Authorization': f"Bearer {self.access_token}"}
+                    response = requests.get(next_page, headers=headers)
+                    body_json = response.json()
+                    next_page = body_json["next_page"]
+                    result = result + body_json['results']
+                return result
+
+            if 400 <= code < 500:
+                raise Exception(body_json['error'])
+
+            raise Exception(f'Unexpected error, code: {code}')
+
+        raise Exception('Workspace with versioning enabled are currently not supported.')
+
+    def list_fields(self, workspace_name: str) -> list:
+        if not self.workspace["isVersioningEnabled"]:
+            version_id = self.workspace['defaultVersionId']
+            params = {'versionId': version_id, 'includeAttributes': 'true'}
+            headers = {'Authorization': f"Bearer {self.access_token}"}
+            response = requests.get(f"{self.url}/fields", params=params, headers=headers)
+            code = response.status_code
+            body_json = response.json()
+            if code == 200:
+                logging.info(
+                    f'list_fields - {len(body_json["results"])} fields found on '
+                    f'workspace: {workspace_name}')
+                result = []
+                result = result + body_json['results']
+                next_page = body_json["next_page"]
+                while next_page is not None:
+                    headers = {'Authorization': f"Bearer {self.access_token}"}
+                    response = requests.get(next_page, headers=headers)
+                    body_json = response.json()
+                    next_page = body_json["next_page"]
+                    result = result + body_json['results']
+                return result
+
+            if 400 <= code < 500:
+                raise Exception(body_json['error'])
+
+            raise Exception(f'Unexpected error, code: {code}')
+
+        raise Exception('Workspace with versioning enabled are currently not supported.')
+
+    def bulk_upsert_sources_tree(self, workspace_name: str, sources: list) -> DataGalaxyBulkResult:
+        # Existing entities are updated and non-existing ones are created.
+        sources_ok_to_bulk = to_bulk_properties_tree(sources)
```

Comment:
I find the naming a bit odd, I would keep it simple:

```suggestion
        bulk_tree = to_bulk_properties_tree(sources)
```